overlord/snapstate/aux_store_info.go

```diff
@@ -0,0 +1,107 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package snapstate
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/snap"
+)
+
+// auxStoreInfo is information about a snap (*not* a snap revision), not
+// needed in the state, that may be stored to augment the information
+// returned for locally-installed snaps
+type auxStoreInfo struct {
+	Media snap.MediaInfos `json:"media,omitempty"`
+}
+
+func auxStoreInfoFilename(snapID string) string {
+	return filepath.Join(dirs.SnapAuxStoreInfoDir, snapID) + ".json"
+}
+
+// retrieveAuxStoreInfo loads the stored per-snap auxiliary store info into the given *snap.Info
+func retrieveAuxStoreInfo(info *snap.Info) error {
+	if info.SnapID == "" {
+		return nil
+	}
+	f, err := os.Open(auxStoreInfoFilename(info.SnapID))
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		}
+		return err
+	}
+	defer f.Close()
+
+	var aux auxStoreInfo
+	dec := json.NewDecoder(f)
+	if err := dec.Decode(&aux); err != nil {
+		return fmt.Errorf("cannot decode auxiliary store info for snap %q: %v", info.InstanceName(), err)
+	}
+	if dec.More() {
+		return fmt.Errorf("cannot decode auxiliary store info for snap %q: spurious content after document body", info.InstanceName())
+	}
+
+	info.Media = aux.Media
+
+	return nil
+}
+
+// retainAuxStoreInfo saves the given auxiliary store info to disk.
+func retainAuxStoreInfo(snapID string, aux *auxStoreInfo) error {
```

Comment:
(nitpick^99 -- SORRY) could we simply call this `saveAuxStoreInfo` or `writeAuxStoreInfo` ? Upon reading more, {load,save,remove}AuxStoreInfo?

Reply:
I proposed these names, given that they are not supposed to be used regularly, and they are kind of subtasks I prefer less plain names

Reply:
I find especially retain slightly obscure but probably just me (being a non-native speaker) but fair enough, its not used much indeed.

Reply:
we could s/retain/keep/ if it's slightly easier ?

I proposed "retain" here with the sense of "keep possession of"

Reply:
we could go `keep` or `stow`

Reply:
stow vs retain doesn't seem a win.  keep is supposedly simple.

Reply:
ok, `keep` works.